.github/workflows/sync_updates_to_oscal_content.yml

```
@@ -0,0 +1,174 @@
+            --verbose --profile --rule --output json 13176 > output.json
+      - name: Test if there are no content changes
+        run: echo "CTF_OUTPUT_SIZE=$(stat --printf="%s" output.json)" >> $GITHUB_OUTPUT
+        id: ctf
+      - name: Print changes to content detected if any
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        run: cat output.json
+
+      # Step 6: Setup the trestle-bot CLI run envrionment
+      - name: Checkout CaC repo
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        uses: actions/checkout@v4
+        with:
+          repository: ComplianceAsCode/content
+          path: content
+      - name: Checkout OSCAL content repo
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        uses: actions/checkout@v4
+        with:
+          repository: huiwangredhat/trestle-repo # TODO: need to update it to OSCAL content repo
+          path: oscal-content
+          token: ${{ secrets.OSCAL_ACCESS_PAT }} # TODO: The access token should be granted the content write permission
+      - name: Checkout trestle-bot repo
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        uses: actions/checkout@v4
+        with:
+          repository: complytime/trestle-bot
+          path: trestle-bot
+      - name: Setup trestlebot
+        run: |
+          cd trestle-bot
+          python3.9 -m pip install --no-cache-dir --upgrade pip setuptools
+          python3.9 -m pip install --no-cache-dir "poetry==1.7.1"
+          poetry install
+
+      # Step 7: Sync CAC content updates to OSCAL content
+      - name: Sync cac content updates to oscal content
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        run: |
+          cd trestle-bot
+          pr_number="${{ github.event.pull_request.number }}"
+          RHEL_PRODUCTS=('rhel8', 'rhel9', 'rhel10', 'ocp4', 'rhcos4', 'rhv4')
+          while IFS= read -r line; do
+            if echo "$line" | grep -q "controls"; then
+              # Sync CAC controls update to OSCAL catalogs, profiles and component-definition
+              CONTROLS=$(echo "$line" | jq -r '.controls[]')
+              for control in $CONTROLS; do
+                if [[ "$control" == *"SRG-APP"* ]] || [[ "$control" == *"SRG-OS"* ]]; then
+                  echo "Ignoring SRG-APP and SRG-OS: $policy_id"
+                  continue
+                fi
+                if [[ "$control" == *"section"* ]]; then
+                  echo "Section found, setting policy_id to 'cis_ocp_1_4_0'"
+                  policy_id="cis_ocp_1_4_0"
+                else
+                  policy_id=$(echo "$control" | cut -f1 -d".")
+                fi
+                poetry run trestlebot sync-cac-content catalog  --repo-path ../oscal-content --committer-email test@redhat.com --committer-name test --branch sync_cac/cac_$pr_number --cac-content-root ../content --policy-id $policy_id --oscal-catalog $policy_id
+                for product in $RHEL_PRODUCTS; do 
```

```suggestion
                for product in "${RHEL_PRODUCTS[@]}"; do
```
This works better with arrays in Bash. I saw you already used something similar bellow so it this one was probably missed. : )

---

The shell syntax is too difficult,  😂